File: 2D renderer.cs
using UnityEngine;
using System.Collections;

public class Render3DModel : MonoBehaviour
{

    string[] args = System.Environment.GetCommandLineArgs();
    if (args.Length > 1)
    {
        objPath = args[1]; // Set the .obj file path from the command line argument
    }
    
    public string objPath = "C:\Users\Mohit\PycharmProjects\Intoxication Detection\face-meshes\5-36-in_mesh.obj"; // Path to your .obj file
    public int imageWidth = 1024; // Width of the output image
    public int imageHeight = 768; // Height of the output image
    public string imagePath = "Assets/renderedImageC#.png"; // Path to save the output image

    void Start()
    {
        // Load the .obj file
        var obj = new OBJLoader().Load(objPath);

        // Add the object to the scene
        Instantiate(obj, Vector3.zero, Quaternion.identity);

        // Create a new RenderTexture
        var rt = new RenderTexture(imageWidth, imageHeight, 24);
        GetComponent<Camera>().targetTexture = rt;

        // Create a new Texture2D and read the RenderTexture into it
        var tex = new Texture2D(imageWidth, imageHeight, TextureFormat.RGB24, false);
        RenderTexture.active = rt;
        tex.ReadPixels(new Rect(0, 0, imageWidth, imageHeight), 0, 0);
        tex.Apply();

        // Convert the Texture2D to a .png
        var bytes = tex.EncodeToPNG();

        // Save the .png
        System.IO.File.WriteAllBytes(imagePath, bytes);
    }
}

File: 2D renderer.py
from UnityEngine import GameObject, Camera, Texture2D, Rect, Quaternion
from System.IO import File
from objreader import ObjReader


class RenderToImage:
    def Start(self):
        # Path to save the rendered image
        imagePath = "RenderedImage.png"
        # Width and height of the rendered image
        imageWidth = 1920
        imageHeight = 1080

        # Load the 3D model from the .obj file
        objFilePath = "face-meshes/5-36-in_mesh.obj"  # Replace "YourModel.obj" with the path to your .obj file
        objReader = ObjReader.read(objFilePath)

        # Create GameObjects for the model's meshes
        for mesh in objReader.meshes:
            gameObject = GameObject.CreatePrimitive(mesh.type)
            gameObject.GetComponent(MeshFilter).mesh = mesh.mesh

        # Set up the camera to render the model
        renderCamera = Camera.main

        # Set camera properties as desired
        # For example, set camera position and rotation
        renderCamera.transform.position = Vector3(0, 1, -10)
        renderCamera.transform.rotation = Quaternion.Euler(0, 180, 0)

        # Render the scene to the render texture
        renderCamera.Render()

        # Create a new texture to store the rendered image
        renderedTexture = Texture2D(imageWidth, imageHeight, TextureFormat.RGB24, False)

        # Read pixels from the render texture and apply to the new texture
        renderedTexture.ReadPixels(Rect(0, 0, imageWidth, imageHeight), 0, 0)
        renderedTexture.Apply()

        # Encode the texture to a PNG file
        bytes = renderedTexture.EncodeToPNG()

        # Save the PNG file
        File.WriteAllBytes(imagePath, bytes)


